lib/tasks/load.rake

```diff
@@ -89,6 +71,29 @@ namespace :db do
     Util::UpdaterV2.new(args).run_main_loop
   end
 
+  desc "Recalculate calculated values for studies in batches"
+  task :recalculate_studies, [] => :environment do
+    batch_size = 5_000
+    total_studies = StudyJsonRecord.version_2.count
+    total_batches = (total_studies / batch_size.to_f).ceil
+    processed_studies = 0
+
+    puts "Starting recalculation for #{total_studies} studies in #{total_batches} batches..."
+
+    (0...total_batches).each do |batch_number|
```

Comment:
You can use the built in method to load records in batches https://api.rubyonrails.org/v7.0.4.2/classes/ActiveRecord/Batches.html#method-i-find_in_batches

Reply:
refactored - thanks for the link!